### ISSUE TITLE
duplicate failed tests result

### ISSUE THREAD

**Issue by danggrianto** [cbeust/testng#319]
using retryanalyzer if i retry a test then the report output is going to report two failed results. In my case sometimes tests failed for unknown reason (accidently) then in this case i only need one result.

**Comment by rschmitt** [comment]
Do you specifically take issue with the "duplicate" results, or with the fact that the unsuccessful test invocations are reported as failures?

**Comment by krmahadevan** [comment]
This is now fixed in the latest released version of `7.1.0`

Sample used

```java
public class Issue319 {

  private int count = 1;

  @Test(retryAnalyzer = TryAgain.class)
  public void testMethod() {
    if (count++ < 2) {
      Assert.fail("Failed");
    }
  }
}
```

Retry analyses used 

```java
import org.testng.IRetryAnalyzer;
import org.testng.ITestResult;

public class TryAgain implements IRetryAnalyzer {

  private int count = 1;

  @Override
  public boolean retry(ITestResult result) {
    return count++ <= 3;
  }
}
```

![image](https://user-images.githubusercontent.com/909773/76695441-6c401f00-66a5-11ea-8636-02acbd722bfb.png)

![image](https://user-images.githubusercontent.com/909773/76695448-837f0c80-66a5-11ea-948b-c9f9e38a6ded.png)

TestNG xml results 

```xml
<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="0" ignored="0" retried="1" total="2" passed="1">
  <reporter-output>
  </reporter-output>
  <suite name="219_suite" duration-ms="61" started-at="2020-03-15T10:08:57 IST" finished-at="2020-03-15T10:08:57 IST">
    <groups>
    </groups>
    <test name="219_test" duration-ms="61" started-at="2020-03-15T10:08:57 IST" finished-at="2020-03-15T10:08:57 IST">
      <class name="com.rationaleemotions.Issue319">
        <test-method status="SKIP" signature="testMethod()[pri:0, instance:com.rationaleemotions.Issue319@6276ae34]" name="testMethod" duration-ms="8" retried="true" started-at="2020-03-15T10:08:57 IST" finished-at="2020-03-15T10:08:57 IST">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Failed]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Failed
at org.testng.Assert.fail(Assert.java:97)
at com.rationaleemotions.Issue319.testMethod(Issue319.java:13)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:134)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:597)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:816)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:748)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testMethod -->
        <test-method status="PASS" signature="testMethod()[pri:0, instance:com.rationaleemotions.Issue319@6276ae34]" name="testMethod" duration-ms="1" started-at="2020-03-15T10:08:57 IST" finished-at="2020-03-15T10:08:57 IST">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testMethod -->
      </class> <!-- com.rationaleemotions.Issue319 -->
    </test> <!-- 219_test -->
  </suite> <!-- 219_suite -->
</testng-results>

```
